src/Illuminate/Database/Schema/PostgresBuilder.php

```diff
@@ -67,7 +67,7 @@ public function dropAllTables()
         foreach ($this->getAllTables() as $row) {
             $row = (array) $row;
 
-            $table = reset($row);
+            $table = '"'.$row['schemaname'].'"."'.$row['tablename'].'"';
```

Comment:
This is a 9.x breaking change for the `$excludedTables` check on line 72. Existing apps with values in  `config('database.pgsql.dont_drop')` won't be schema-qualified with wrapping quotes. Even the hardcoded `$excludedTables = ['spatial_ref_sys']` default value used by 99.9% of Laravel/Postgres apps will fail to be skipped.

Reply:
Great catch! Thank you! This is exactly why I said the following in the related Issue:

> As is the case with all things search_path-related, the fix is likely to be nuanced and will have to be vetted carefully so as not to break anything else, as not all of the affected code has test coverage (particularly the older code).

I'll work on a solution to this. Thanks again!

Reply:
Whenever it's convenient, please take a look at the new commit that I pushed. My hope is that it resolves the concern that you raise here.

Now, tables are excluded in a backwards-compatible way, so that all of the following table notations would be excluded:

```php
'dont_drop' => [
    'users',
    '"password_resets"',
    'public.failed_jobs',
    '"public"."personal_access_tokens"',
    'spatial_ref_sys'
],
```

This configuration produces the following `DROP` query in a stock Laravel installation in which `php artisan migrate` has been run (and `migrations` is the only table we _didn't_ exclude):

```
"drop table "public"."migrations" cascade"
```

So, in essence, the user can schema-qualify those `dont_drop` references if desired, and if not, the unqualified references will be treated just as they have been prior to this PR. That is to say, when a reference is unqualified, the table will not be dropped from _any_ schema in which it exists.

As a related aside, the the current logic around the `spatial_ref_sys` table that exists when the `postgis` extension is enabled  seems a little odd to me.

In essence, the logic is, "If the user has _not_ specified tables to be excluded, then exclude `spatial_ref_sys`, else, exclude the specified tables (in which case `spatial_ref_sys` will be dropped)."

Is that necessarily the desired behavior? It seems all too easy to drop `spatial_ref_sys` without even realizing it's happening, merely by specifying some other, completely unrelated table(s) to be excluded. I suppose once the user realizes what's happening, they'll add `spatial_ref_sys` to the `dont_drop` array, too.

The same is true of the `dropAllViews()` method. It'll drop the two views that are created for `postgis` (`geography_columns` and `geometry_columns`):

```
laravel=# CREATE EXTENSION postgis;            
CREATE EXTENSION                               
laravel=# \d                                   
               List of relations               
 Schema |       Name        | Type  |  Owner   
--------+-------------------+-------+----------
 public | geography_columns | view  | postgres 
 public | geometry_columns  | view  | postgres 
 public | spatial_ref_sys   | table | postgres 
(3 rows)                                       
```

I don't know if any of this is of concern to anybody, I just figured I'd mention it while on the subject.